src/round.rs

```diff
@@ -166,10 +184,19 @@ impl<Id: Hash + Eq + Clone, Vote: Clone + Eq, Signature: Clone> VoteTracker<Id,
 					*occupied.get_mut() = new_val;
 				}
 
-				&*occupied.into_mut()
+				Some(&*occupied.into_mut())
 			}
 		}
 	}
+
+	fn valid_votes<'a>(&'a self) -> impl Iterator<Item=(Id, Vote, Signature)> + 'a {
+		self.votes.iter().filter_map(|(id, vote)| {
+			match vote {
+				VoteMultiplicity::Single(v, s) => Some((id.clone(), v.clone(), s.clone())),
```

Comment:
I'm not sure what the repercussions of ignoring equivocations will be here. We may end up in situations where we have finalized blocks with equivocations, but cannot craft a commit message.